internal/server/server.go

```diff
@@ -108,45 +109,38 @@ func New(opts ...ServerFuncOpt) (*Server, error) {
 	return s, nil
 }
 
-func httpClientForRootCAs(rootCABytes []byte) (*http.Client, error) {
-	tlsConfig := tls.Config{RootCAs: x509.NewCertPool()}
-	if !tlsConfig.RootCAs.AppendCertsFromPEM(rootCABytes) {
-		return nil, fmt.Errorf("no certs found in rootCABase64")
+func (s *Server) initHttpClient() error {
```

Comment:
I refactored this a bit so that it is easier to call in `handleHealthCheck()`.
Functionally it is the same as before, with `transport.DisableKeepAlives = true` being the change.

Reply:
Is disable keep alive still necessary if we are going to create a new http.Client with new transport?

Reply:
Yeah thinking here is DisableKeepAlives=true is the first solution that I'm trying. If the lockup issue never happens again, then great, this is the solution.
If lockup happens again, then we fall back to creating a new http.Client.
I'm trying DisableKeepAlives first since it seems like a "cleaner" solution. It's more of a setting rather than us forcefully creating a new http.Client each time which may be seen as wrong code-wise I guess.